lib/winston/logger.js

```diff
@@ -647,7 +634,9 @@ class Logger extends Transform {
 
   _addDefaultMeta(msg) {
     if (this.defaultMeta) {
-      Object.assign(msg, this.defaultMeta);
+      // The msg must be cloned as it is being mutated, but any metadata provided with the msg takes precedence over default
+      const msgClone = JSON.parse(jsonStringify(msg));
```

Comment:
Super late to the party, but because of this change there is no possibility of logging instances of error objects in any way. All the previous suggestions of adding custom formatters in order to extract separate parts of the error object are not applicable anymore. By the time any formatter is called, the error object was already stringified and parsed back, thus losing message, stack, etc. props. As of now, a user of this library should choose: be able to log errors or use defaultMeta logger property.

Reply:
This looks to be the case. I was under the impression at the time that the ErrorFormatter addressed the use case sufficiently. Obviously I was wrong in that respect. Hence why we did a patch release to "revert" the regression.

I'm going to be working through these scenarios this week ideally to fix the regression. Any test cases you have would be hugely appreciated!!

Reply:
Are there any side-effects if I follow this approach?

`const msgClone = JSON.parse(jsonStringify(msg));
 Object.assign(msg, this.defaultMeta, msgClone);
`

cause I am facing the issue where defaultMeta is overriding my msg data.